lib/manageiq/integration/foreman_manager.rb

```diff
@@ -0,0 +1,106 @@
+require "fileutils"
+
+# A wrapper around the foreman utility for running ManageIQ in dev mode for
+module ManageIQ
+  module Integration
+    class ForemanManager
+      def self.pid_file
+        @pid_file ||= Environment.tmp_dir.join('foreman.pid')
+      end
+
+      def self.log_file
+        Rails.root.join("log", "integration.foreman.log")
+      end
+
+      def self.procfile
+        @procfile ||= Environment.tmp_dir.join('Procfile')
+      end
+
+      def self.example_procfile_data
+        require 'erb'
+        template_path = File.expand_path('Procfile.example.erb', __dir__)
+
+        ERB.new(File.read(template_path)).result(binding)
+      end
+
+      def self.setup
+        FileUtils.mkdir_p(Environment.tmp_dir)
+        FileUtils.rm_f(procfile)
+
+        File.write(procfile, example_procfile_data)
+      end
+
+      def self.run
+        if process_alive?
+          warn "foreman process already running"
+        else
+          foreman_cmd = %W[
+            foreman start --port=3000
+                          --root=#{Rails.root}
+                          --procfile=#{procfile}
+          ].join(" ")
+
+          pid = Process.spawn(foreman_cmd, [:out, :err] => [log_file, "a"])
+          Process.detach(pid)
+          File.write(pid_file, pid)
+        end
+      end
+
+      def self.stop
+        if server_pid
+          Process.kill("TERM", File.read(pid_file).to_i)
+          FileUtils.rm(pid_file)
+        else
+          warn "PID file for server (#{pid_file}) doesn't exist... moving on"
+        end
+      end
+
+      # Simple status check:
+      #
+      # - Checks for existing pid file
+      # - If the process is running
+      # - If the UI is responding to a request
+      #
+      def self.status
+        if server_pid
+          puts "Server PID##{server_pid}: #{process_alive? ? 'running' : 'stopped'}"
+
+          puts "UI running?:      true" if process_alive?
+        else
+          warn "PID file for server (#{pid_file}) doesn't exist..."
+        end
+      rescue RuntimeError # error from .ui_running?
+        puts "UI running?:      false"
+      end
+
+      def self.process_alive?
+        server_pid && Process.getpgid(server_pid).to_i > 0
+      rescue
+        false
+      end
+
+      def self.server_pid
+        File.read(pid_file).to_i if File.exist?(pid_file)
+      end
+
+      def self.ui_running?
+        require 'net/http'
+
+        ui_endpoint  = URI("http://localhost:3000/api/ping")
```

Comment:
The 3000 here and as the argument to foreman could be made a constant above, in case we find the need to ever need multiple puma workers.